mhs/mhs/common/messages/tests/test_ebxml_ack_envelope.py

```diff
@@ -61,11 +65,28 @@ def test_from_string(self):
 
         parsed_message = ebxml_ack_envelope.EbxmlAckEnvelope.from_string({}, message)
 
-        self.assertEqual(test_ebxml_envelope.EXPECTED_VALUES, parsed_message.message_dictionary)
+        self.assertEqual(EXPECTED_VALUES, parsed_message.message_dictionary)
 
     def test_from_string_with_no_values(self):
         message = file_utilities.FileUtilities.get_file_string(str(self.message_dir / "ebxml_header_empty.xml"))
 
-        parsed_message = ebxml_ack_envelope.EbxmlAckEnvelope.from_string({}, message)
+        with self.assertRaisesRegex(ebxml_envelope.EbXmlParsingError, "Weren't able to find required element"):
+            ebxml_ack_envelope.EbxmlAckEnvelope.from_string({}, message)
+
+    @patch.object(message_utilities.MessageUtilities, "get_timestamp")
+    @patch.object(message_utilities.MessageUtilities, "get_uuid")
+    def test_roundtrip(self, mock_get_uuid, mock_get_timestamp):
```

Comment:
The reason I haven't added a roundtrip test to `test_ebxml_request_envelope.py` is because on the initial request, we don't send a `RefToMessageId`, but on the inbound request from Spine, we do get a `RefToMessageId` element (this makes sense as we send only the initial message id on our request to Spine, and when Spine sends a request back to us, it sends us a message id for the individual request and the message id we sent in previously). It's not that it wouldn't have been possible, but it would have required me to fiddle with the MIMEd xml message string.